ArduPlane/sensors.cpp

```diff
@@ -77,6 +79,27 @@ void Plane::read_airspeed(void)
     if (ahrs.airspeed_estimate(&aspeed)) {
         smoothed_airspeed = smoothed_airspeed * 0.8f + aspeed * 0.2f;
     }
+
+    if(ahrs.airspeed_sensor_enabled()) {
+        float wind_max = ahrs.get_max_wind();
+        //gcs().send_text(MAV_SEVERITY_INFO, "Max Wind: %f", wind_max);
+        //gcs().send_text(MAV_SEVERITY_INFO, "Airspeed: %f", aspeed);
+
+        if (wind_max > 0 && AP::gps().status() >= AP_GPS::GPS_OK_FIX_2D) {
+            float gnd_speed = AP::gps().ground_speed();
+            //gcs().send_text(MAV_SEVERITY_INFO, "Ground Speed: %f", gnd_speed);
+            // return false as the airspeed is not as expected and could be faulty
+            if(aspeed > (gnd_speed + wind_max) || aspeed < (gnd_speed - wind_max)) {
+                airspeed_failed_count++;
+
+                if(airspeed_failed_count > 30) {
+                    gcs().send_text(MAV_SEVERITY_WARNING, "Airspeed sensor - possibly faulty - ARSPED_USE: 0");
```

Comment:
I think the warning should be a little different. Thoughts on

"Airspeed fault - AHRS_WIND_MAX exceeded - airspeed use disabled"

Reply:
Agreed I'll make change.